src/internal/pfsdb/branches.go

```diff
@@ -44,28 +56,39 @@ const (
 	BranchColumnUpdatedAt = branchColumn("branch.updated_at")
 )
 
-// SliceDiff takes two slices and returns the elements in the first slice that are not in the second slice.
-// TODO this can be moved to a more generic package.
-func SliceDiff[K comparable, V any](a, b []V, key func(V) K) []V {
-	m := make(map[K]bool)
-	for _, item := range b {
-		m[key(item)] = true
-	}
-	var result []V
-	for _, item := range a {
-		if !m[key(item)] {
-			result = append(result, item)
-		}
+type ErrBranchProvCycle struct {
```

Comment:
Can we add a comment to this like ErrBranchNotFound and make sure it has a GRPC error code if we need to use it on the client.